src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -13,15 +8,16 @@ namespace Orleans.Serialization
 {
     internal class OrleansJsonSerializer : IExternalSerializer
     {
-        public static JsonSerializerSettings settings;
+        internal static JsonSerializerSettings SerializerSettings { get; private set; }
+
         private TraceLogger logger;
 
         static OrleansJsonSerializer()
         {
-            settings = new JsonSerializerSettings
+            SerializerSettings = new JsonSerializerSettings
             {
                 TypeNameHandling = TypeNameHandling.All,
-                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                PreserveReferencesHandling = PreserveReferencesHandling.All,
```

Comment:
I think you misunderstood the intent.
The intent is one line: AzureTableStorage.cs, line 106.